src/lib/Core.hs

```diff
@@ -324,6 +325,10 @@ nestToEnvFrag = nestToEnvFragRec emptyOutFrag
 instance BindsEnv b => (BindsEnv (NonEmptyNest b)) where
   toEnvFrag (NonEmptyNest b rest) = toEnvFrag $ Nest b rest
 
+instance BindsEnv b => (BindsEnv (RNest b)) where
```

Comment:
Why is this instance so much simpler than the corresponding instance for Nest?

Reply:
I rebased an older commit that I wrote before I optimized some Nest functions. But some of those Nest optimizations are not necessary anymore, thanks to making InplaceT be a state monad in decls, not a writer (implementation-wise). I ported the ones that might still be useful to the RNest case.